src/lint.ts

```diff
@@ -45,7 +45,8 @@ export interface LintResult {
 }
 
 export interface ILinterOptions {
-    configuration: any;
+    configuration?: any;
+    configurationPath?: string;
```

Comment:
Most of the changes in this PR are good, but I'm wondering if we actually need this `configurationPath` option. If 3rd-party tools use `findConfiguration/loadConfigurationFromPath`, those functions can resolve relative paths which the programatic API should then be able to handle.